daemon/cmd/app_logger.go

```diff
@@ -0,0 +1,132 @@
+package cmd
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"go.uber.org/fx/fxevent"
+
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+type appLogger struct {
+	*logrus.Entry
+}
+
+func newAppLogger() fxevent.Logger {
+	return appLogger{
+		logging.DefaultLogger.WithField(logfields.LogSubsys, "daemon-app"),
```

Comment:
Is this deliberately exposing these logs as a different subsystem from `daemonSubsys`? So far I've mostly assumed that a "subsystem" is for the most part a specific directory. For all the daemon stuff we just use `daemon` but for all the various `pkg/foo/bar` we typically name these with package-level names like `foo-bar` matching the path. This way, when there's a log, you can clearly see which package the log came from by looking at the subsys field.
```suggestion
		logging.DefaultLogger.WithField(logfields.LogSubsys, daemonSubsys),
```

Reply:
Yes I was being deliberate in doing that in order to be able to find logs related to fx itself, but your reasoning with sticking to package names makes sense. Perhaps there's another field we could use for this?